pkg/v1/daemon/options.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+func WithBufferedOpener() ImageOption {
+	return func(i *ImageOpener) error {
+		return i.SetBuffered(true)
+	}
+}
+
+func WithUnbufferedOpener() ImageOption {
+	return func(i *ImageOpener) error {
+		return i.SetBuffered(false)
+	}
+}
+
+func (i *ImageOpener) SetBuffered(buffer bool) error {
```

Comment:
Generally the same comments as in remote.Image here.  Can you see if we can make `ImageOpener` private and primarily accessible thru `WithFoo` passed to `blah.Image`?